### PR TITLE
Dockerize

### PR DESCRIPTION
**Pull request by arvindkandhare**
Dockerfile to create a docker image. This image runs bookie and writeproxy.

### CODE REVIEW

**Comment by franckcuny** [comment]
@arvindkandhare regarding docker, I've started https://github.com/franckcuny/docker-distributedlog, would you mind collaborating there ? 

I personally prefer to keep Dockerfiles outside of the repo. Vagrant makes sense, since it's mostly used for development purpose, and it's easy to re-use across teams.

@sijie do you have an opinion ?

**Comment by arvindkandhare** [comment]
> would you mind collaborating there

Hi,
Sure. I can roll either ways :)
For completeness sake,   here is a possible argument for doing the opposite. If the code is going to change dramatically and the Dockerfile needs to change with it, then it makes sense to have it in the same repo. 
The other point is that I am seeing this trend where a lot of projects are moving their Dockerfile out of the tree and in a separate repo. 

One more point I noticed is that you save separate dockerfiles for deploying zk,bk and DL (writeproxy). Looking at it from perspective of teams who have not used BK before, I think it makes sense to put BK and writeproxy together(A global ZK will be used per config). I think most of the deployments would already have it deployed.
Again, this is just a point of view for completeness. I see the need to do it the way you are doing it and appreciate it,too. 
@sijie Do let us know your point of view here.

**Comment by sijie** [comment]
**Re: Dockerfile repo**

I am not sure what is the best practice for maintaining the docker files. will docker file keep changing with each dl changes or release? for example, a new release.

I would prefer whatever is easier for users and releases. but @franckcuny @arvindkandhare I will let you guys decide.

**Re: Docker image**

personally I would prefer a single image for bk & write-proxy and have a flag to control whether to start bk, or start write proxy, or start both. Having single image will reduce the maintenance overhead.

From deployment wise, we can still have two ways to deployment write proxy and bk.

- "Broker"-like way: use single image to run bk & write-proxy together.
- 'Micro-service'/'Distributed' way: deploy storage first using same image to start bk only, then deploy write proxy using same image but only start write proxy.

I do see the advantages for both deployments:

_- "Broker"_: it is easy to run for a small cluster, would be useful for most of the use cases.
_- "Distributed"_: it is easy to manage in a large scale deployment, as 'storage' and 'write-proxy' can scale independently

So I would like you guys to explore if we can support both using single image.

**Re. other comment:**

Also I would prefer adding document about how to run or deploy to any cloud provider using docker image. That would be super helpful.

**Comment by arvindkandhare** [comment]
Hi,
I have updated the pull request as per our discussions. Let me know your feedback.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+FROM java:8
+
+MAINTAINER Arvind Kandhare (arvind.kandhare@emc.com)
```
Comment (franckcuny) [inline Dockerfile]: replace () with <>

**Comment by franckcuny** [comment]
Except for a minor comment, this LGTM.

**Comment by arvindkandhare** [comment]
> LGTM

Hi,
I have updated the pull request with this comment taken care of. Please pull if there are no further comments.

**Comment by franckcuny** [comment]
@arvindkandhare Thanks!

**Comment by sijie** [comment]
@franckcuny @arvindkandhare what do you guys think about adding documentation about running docker image?

**Comment by arvindkandhare** [comment]
Yes. I will take it up. This is my ToDo bucket. Was feeling a bit lazy :). Do share checklist etc about how to create documentation (privately) and will get started on it.

**Comment by sijie** [comment]
@arvindkandhare 

The docs are under the `docs` directory. It is rewritten in [reStructuredText](http://docutils.sourceforge.net/rst.html) syntax. However, the documentation and website are generated using an internal tool. When we are moving to Apache, we are looking for any solutions to build the documentation and website. But you can go ahead to make changes to the docs under docs directory.

**Pull request merged**